### ISSUE TITLE
Refer Ubuntu image build error #9715

### ISSUE THREAD

**Issue by jaswant-dnv** [actions/runner-images#9788]
### Description

Refer Ubuntu image build error #9715

https://github.com/actions/runner-images/issues/9715

### Platforms affected

- [X] Azure DevOps
- [ ] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [X] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [ ] macOS 13
- [ ] macOS 13 Arm64
- [ ] macOS 14
- [ ] macOS 14 Arm64
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

Image 20240422.1.0

### Is it regression?

N/A

### Expected behavior

Image build completes

### Actual behavior

Refer Ubuntu image build error #9715

### Repro steps

rerun pipeline

**Comment by erik-bershel** [comment]
Hey there!

It's a duplicate of this item: https://github.com/actions/runner-images/issues/9708
Please check your firewall configuration to allow `hkp` connection.

**Comment by LokeshNagireddy** [comment]
it worked with previous build which used the image 3afd153301d1a498c8d2be9c7abfed7cfa8e5498, but not working with current commit.

Can you please reverify.

**Comment by erik-bershel** [comment]
@LokeshNagireddy it's different issue. There is no connection between `hpk` and `software report` blockers.

**Comment by LokeshNagireddy** [comment]
@erik-bershel , Can you please let me know whether hpk is required to install swift? i changed to latest image reference, but still we have the same error.

2024-05-07T11:19:05.6121475Z ==> azure-arm.build_image: Provisioning with shell script: /azp/_work/1/s/runner-images/images/ubuntu/templates/../scripts/build/install-swift.sh
2024-05-07T11:19:05.9619990Z ==> azure-arm.build_image: Downloading package from https://swift.org/builds/swift-5.10-release/ubuntu2204/swift-5.10-RELEASE/swift-5.10-RELEASE-ubuntu22.04.tar.gz to /tmp/swift-5.10-RELEASE-ubuntu22.04.tar.gz...
2024-05-07T11:19:13.1929077Z ==> azure-arm.build_image: Package downloaded in 8 seconds
2024-05-07T11:19:13.2521684Z ==> azure-arm.build_image: gpg: directory '/root/.gnupg' created
2024-05-07T11:19:13.2526865Z ==> azure-arm.build_image: gpg: **keybox '/root/.gnupg/pubring.kbx' created**
2024-05-07T11:26:44.4047542Z ==> azure-arm.build_image: gpg: **keyserver receive failed: Connection timed out**

i dont understand what im missing.. is there any update in image happened after April 10th?
im using this image for linux

**Comment by LokeshNagireddy** [comment]
@erik-bershel, May you please clarify this issue. why swift installation working with some commit, and not working with latest commit. Surely you changed something after the commit https://github.com/actions/runner-images/commit/3afd153301d1a498c8d2be9c7abfed7cfa8e5498.

I look forward for your response.

**Comment by erik-bershel** [comment]
Hey @LokeshNagireddy!

Swift installation guide: https://www.swift.org/install/linux/#Installation-via-Tarball
Successful Ubuntu-20.04 nightly build: [20240508.1](https://dev.azure.com/mseng/AzDevNext.Deploy/_build/results?buildId=28089150&view=results)
Successful Ubuntu-22.04 nightly build: [20240508.1](https://dev.azure.com/mseng/AzDevNext.Deploy/_build/results?buildId=28089157&view=results)

Your problem is related to the configuration of your Azure org. Which resulted in the fact that the port required for `hkp` connection is forbidden by org's firewall. At the same time, we have no problem creating images, as you can see from the links provided. You have to options: whether to update your org's configuration or to use `hkps` in your fork.

To summarise, I can say that the problem is not related to our repository and will not be solved here. We will happily change the connection type to `hkps` as soon as the Swift developers make this the recommended method.

**Comment by arashjalalat** [comment]
@erik-bershel We opened the default hkp port 11371 which resolved the 'Connection timed out' issue, however I'm now facing the error message: 'gpg: keyserver receive failed: End of file'. I followed the Swift installation guide: https://www.swift.org/install/linux/#Installation-via-Tarball and noticed that if I add port 80 to the endpoint the PGP keys are imported correctly. 

```
gpg --keyserver hkp://keyserver.ubuntu.com:80 \
      --recv-keys \
      '7463 A81A 4B2E EA1B 551F  FBCF D441 C977 412B 37AD' \
      '1BE1 E29A 084C B305 F397  D62A 9F59 7F4D 21A5 6D5F' \
      'A3BA FD35 56A5 9079 C068  94BD 63BC 1CFE 91D3 06C6' \
      '5E4D F843 FB06 5D7F 7E24  FBA2 EF54 30F0 71E1 B235' \
      '8513 444E 2DA3 6B7C 1659  AF4D 7638 F1FB 2B2B 08C4' \
      'A62A E125 BBBF BB96 A6E0  42EC 925C C1CC ED3D 1561' \
      '8A74 9566 2C3C D4AE 18D9  5637 FAF6 989E 1BC1 6FEA' \
      'E813 C892 820A 6FA1 3755  B268 F167 DF1A CF9C E069'
```

**Comment by arashjalalat** [comment]
@erik-bershel Will the [Swift install script](https://github.com/actions/runner-images/blob/main/images/ubuntu/scripts/build/install-swift.sh) be modified to use port 80?

**Comment by erik-bershel** [comment]
Hey @arashjalalat!

We try not to change underlying ports or protocols unless it is directly recommended by the developers or is the only method available. In this case, I don’t see much point in changing the port to 80 - this may block other users and is generally unsafe. You can use it as a workaround in your fork the same time.
